experimental/InvariantTheory/src/TorusInvariantsFast.jl

```diff
@@ -264,14 +262,9 @@ julia> fundamental_invariants(RT)
  X[2]^2*X[3]
 ```
 """
-function fundamental_invariants(z::TorGrpInvRing)
-    if isdefined(z, :fundamental)
-        return z.fundamental
-    else
-        R = z.representation
-        z.fundamental = torus_invariants_fast(weights(R), polynomial_ring(z))
-        return z.fundamental
-    end
+@attr function fundamental_invariants(z::TorGrpInvRing)
```

Comment:
```suggestion
@attr SomeType function fundamental_invariants(z::TorGrpInvRing)
```
Please add the return type here to make the attribute lookup type stable

Reply:
I don't know what type to use here other than Vector{MPolyRingElem}  or Vector{MPolyDecRingElem} - which are not concrete types.

Reply:
Faced the same issue with the other file - InvariantTheory.jl

Reply:
That's true. In most other places, we would add the field type as a type parameter to the invariant ring, which would then allow something smarter here.
But I think this is out of scope for this PR. 

@fingolfin can you keep this in mind for some future refactoring?

Reply:
Yes, one would have to add parameters to the type `TorGrpInvRing` to make this more precise. I can have a look; I wanted to rename the types anyway in #3442.

Reply:
Sorry, I typed to slow. We can keep this here as is and I look into it in #3442.

Reply:
@lgoettgens sure, I actually had pointed out the same issue on a prior PR. We'll get there eventually ;-)